azure-kit-starter/src/main/java/com/vaadin/azure/starter/ui/ClusterSupport.java

```diff
@@ -0,0 +1,245 @@
+    public static class VersionNotificator extends Component {
+        private static final long serialVersionUID = 1L;
+
+        VersionNotificator(String current, String update) {
+            getElement().setProperty("currentVersion", current);
+            getElement().setProperty("updateVersion", update);
+        }
+
+        public Registration addSwitchVersionEvent(
+                ComponentEventListener<SwitchVersionEvent> listener) {
+            getLogger().debug("Adding listener for SwitchVersionEvent.");
+            return addListener(SwitchVersionEvent.class, listener);
+        }
+
+        @DomEvent("load-version")
+        public static class SwitchVersionEvent
+                extends ComponentEvent<VersionNotificator> {
+            private static final long serialVersionUID = 1L;
+
+            public SwitchVersionEvent(VersionNotificator source,
+                    boolean fromClient) {
+                super(source, fromClient);
+            }
+        }
+    }
+
+    /**
+     * Interface for receiving events on cluster change.
+     */
+    public interface SwitchVersionListener extends Serializable {
+
+        /**
+         * Notify about the cluster node change to allow graceful transition of
+         * the users. <b>Note:</b> Even returning <code>false</code>, the
+         * application might still be shut down by the environment.
+         *
+         * @param vaadinRequest
+         *            Vaadin request when the change is initiated.
+         * @param vaadinResponse
+         *            Response from the server to be sent to the client.
+         * @return <code>@code true</code> if the cluster change is ok, false if
+         *         the change should not be performed.
+         */
+        boolean clusterSwitch(VaadinRequest vaadinRequest,
+                VaadinResponse vaadinResponse);
+    }
+
+    /**
+     * Register the version switch listener.
+     *
+     * @param listener
+     *            SwitchVersionListener to register.
+     */
+    public void setSwitchVersionListener(SwitchVersionListener listener) {
+        this.switchVersionListener = listener;
+    }
+
+    private Cookie getCookieByName(String cookieName) {
+        VaadinRequest request = VaadinRequest.getCurrent();
+        if (cookieName == null || request == null
+                || request.getCookies() == null) {
+            return null;
+        }
+        for (Cookie c : request.getCookies()) {
+            if (cookieName.equals(c.getName())) {
+                return c;
+            }
+        }
+        return null;
+    }
+
+    private void removeCookie(String cookieName) {
+        getLogger().debug("Removing cookie '{}'.", cookieName);
+        Cookie cookie = new Cookie(cookieName, "");
+        cookie.setMaxAge(0);
+        VaadinResponse.getCurrent().addCookie(cookie);
+    }
+
+    public static ClusterSupport getCurrent() {
+        return CurrentInstance.get(ClusterSupport.class);
+    }
```

Comment:
What's the use case of getting a `ClusterSupport` instance as public API?

Reply:
The `setSwitchVersionListener()` function can be called on it.

Reply:
Just a remark from the previous project: Maybe a single listener is not even enough, but the use case is to do application level clean-up before version switch and possibly also possibility to cancel the dialog in case application wants to force version switch.

Reply:
@samie Added a function to have the possibility to do application level cleanup before version switch. The force canceling of the dialog will happen if we remove the previous version of the application from the cluster and force the new version by k8s manifests and commands.